pkg/controllers/dynakube/controller.go

```diff
@@ -60,16 +62,22 @@ func NewController(mgr manager.Manager, clusterID string) *Controller {
 
 func NewDynaKubeController(kubeClient client.Client, apiReader client.Reader, scheme *runtime.Scheme, config *rest.Config, clusterID string) *Controller { //nolint:revive
 	return &Controller{
-		client:                 kubeClient,
-		apiReader:              apiReader,
-		scheme:                 scheme,
-		fs:                     afero.Afero{Fs: afero.NewOsFs()},
-		dynatraceClientBuilder: dynatraceclient.NewBuilder(apiReader),
-		istioClientBuilder:     istio.NewClient,
-		registryClientBuilder:  registry.NewClient,
-		config:                 config,
-		operatorNamespace:      os.Getenv(env.PodNamespace),
-		clusterID:              clusterID,
+		client:                              kubeClient,
+		apiReader:                           apiReader,
+		scheme:                              scheme,
+		fs:                                  afero.Afero{Fs: afero.NewOsFs()},
+		config:                              config,
+		operatorNamespace:                   os.Getenv(env.PodNamespace),
+		clusterID:                           clusterID,
+		dynatraceClientBuilder:              dynatraceclient.NewBuilder(apiReader),
+		istioClientBuilder:                  istio.NewClient,
+		registryClientBuilder:               registry.NewClient,
+		deploymentMetadataReconcilerBuilder: deploymentmetadata.NewReconciler,
```

Comment:
I'm trying understand what is purpose of adding these 5 lines between 75-80:
For example `istioReconcilerBuilder`, first of all we pass only clients here, not instances of reconcilers. Controller struct implements the Reconciler interface. Why do we pass reconcilers as input?

Also quick grep demonstrates nothing to me:
```
rg istioReconcilerBuilder
pkg/controllers/dynakube/controller_system_test.go
84:             istioReconcilerBuilder:              istio.NewReconciler,

pkg/controllers/dynakube/controller.go
79:             istioReconcilerBuilder:              istio.NewReconciler,
115:    istioReconcilerBuilder              istio.ReconcilerBuilder
264:    istioReconciler := controller.istioReconcilerBuilder(istioClient)

pkg/controllers/dynakube/controller_test.go
692:                    istioReconcilerBuilder: istio.NewReconciler,
```